### ISSUE TITLE
IsomiRs package error on IsomirDataSeqFromFiles function

### ISSUE THREAD

**Issue by trachtok** [bioconda/bioconda-recipes#26558]
Hi,

I have problem running Bioconductor isomiRs package using conda. Specifically version 1.14, 1.16, 1.18 (tried all of them), previous versions (1.8, 1.10) work ok. When I install isomiRs locally on my laptop without using conda (version 1.16) it runs ok, using conda (on our server) I end with an error when running function `IsomirDataSeqFromFiles()` that should just read input files into an object (and without it, I cannot move to anything else). Error message:

```
Error in (function (cond)  : 
  error in evaluating the argument 'x' in selecting a method for function 'as.matrix': `.data` must be a data frame without row names.
Calls: IsomirDataSeqFromFiles ... cnd_signal -> signal_abort -> signalCondition -> <Anonymous>
In addition: There were 50 or more warnings (use warnings() to see the first 50)
Execution halted
```
It seems that some dependency package that handles tables/data frames is not of proper version within conda isomiRs package and it causes the error on initial loading of files. The command I used to create conda env with isomiRs:

`conda create -n isomirs116 -c bioconda -c conda-forge -c r bioconductor-isomirs=1.16.0`

Also, when I load isomiRs library in this created conda env I get the following warning:
```
Warning message:
package ‘matrixStats’ was built under R version 4.0.3
```
 When I try to install isomiRs AND r-base=4.0.3 I got a following conflict:
```
Package r-base conflicts for:
bioconductor-isomirs=1.16.0 -> r-base[version='>=4.0,<4.1.0a0']
r-base=4.0.3
bioconductor-isomirs=1.16.0 -> r-assertive.sets -> r-base[version='3.1.3.*|3.2.0.*|3.2.1.*|3.2.2.*|3.3.1.*|3.3.2.*|3.4.1.*|>=3.4.1,<3.4.2.0a0|>=3.5,<3.6.0a0|>=3.6,<3.7.0a0|>=3.5.1,<3.5.2.0a0|>=3.5.0,<3.5.1.0a0|>=3.4.3,<3.4.4.0a0|>=3.4.2,<3.4.3a0']
```

I have put input test files (made by miraligner 3.2), R script that reads these input files and runs function `IsomirDataSeqFromFiles()` (requires path to input files as a parameter) and a list of packages in conda env created by above-written command into a [dropbox](https://drive.google.com/drive/u/1/folders/107Du6BwaQtbDnFYUmDfNZcutv4x7aKAW). Unfortunately, isomiRs have quite a lot of dependencies and I was not able to find out what is the exact problem. I tried to create different conda env on multiple unrelated computers with different debian/ubuntu/macos and I get this error everywhere, so I think it is problem within conda package, because installation of isomiRs on my local PC within RStudio (without conda) works ok. 

Thank you!
Karolina

**Comment by karl616** [comment]
I see something similar, but with `IsomirDataSeqFromMirtop`. It seems to come down the dplyr chain in this function: `isomiRs:::IsoCountsFromMatrix`. This function is called by `IsomirDataSeqFromRawData`, which in turn is referred by the two problematic functions. Towards the end of the chain there is a `column_to_rownames("uid")` call. Changing that to `remove_rownames() %>% column_to_rownames("uid")` might do the trick...

**Comment by zxl124** [comment]
I can confirm I am having the same problem in a conda environment built recently. A different conda environment which was built a few months ago with the same version of `isomiRs` does not have this bug. From what @karl616 suggested, I suspected a newer build/version of either `dplyr` or `tibble` is the culprit. After comparing the two environments and experimenting rolling back `dplyr` and `tiblle`, I found downgrading `dplyr` from `r-dplyr=1.0.4=r40h03ef668_0` to `r-dplyr=1.0.2=r40h0357c0b_0` eliminates this bug. I haven't tested for other `isomiRs` functions, By the way, I am using `IsomirDataSeqFromMirtop`.

I haven't looked at `isomiRs` code to see if this is suboptimal coding on the part of `isomiRs` or something was wrong with the new `r-dplyr` conda build. It is strange that this problem only applies to conda-built `isomiRs`.

**Comment by karl616** [comment]
There was quite a few updates to the functions running prior to column_to_rownames with dplyr 1.0.3 (https://github.com/tidyverse/dplyr/blob/master/NEWS.md). I would guess that something happened there, inducing this. And without confirming it, I suspect the error is present in a non-conda installation... conda environments are just created and updated much more frequently ==> statistics ==> consequences

**Comment by trachtok** [comment]
I also agree with @karl616, I am running isomiRs 1.16 locally with dplyr 1.0.2 and it works. Using a higher dplyr version does not work. So what would be the best way of solving this? :) Making conda isomiRs package for now use only lower version of dplyr and contact authors of isomiRs with info that the new version of dplyr does not seem to work with their code? I checked issues on their github page and there does not seem to be one about this problem, so they might not be aware of it.
